edge/pkg/devicetwin/dtmanager/twin_test.go

```diff
@@ -0,0 +1,1534 @@
+		t.Run(test.name, func(t *testing.T) {
+			if err := dealUpdateResult(test.context, test.deviceID, test.eventID, test.code, test.err, test.payload); !reflect.DeepEqual(err, test.errWant) {
+				t.Errorf("DTManager.TestDealUpdateResult() case failed: got = %v, Want = %v", err, test.errWant)
+			}
+		})
+	}
+}
+
+// TestDealDelta is function to test dealDelta
+func TestDealDelta(t *testing.T) {
+	tests := []struct {
+		name     string
+		context  *dtcontext.DTContext
+		deviceID string
+		payload  []byte
+		errWant  error
+	}{
+		{
+			name:     "TestDealDelta(): Case 1",
+			context:  &dtcontext.DTContext{},
+			deviceID: "DeviceA",
+			payload:  []byte(""),
+			errWant:  errors.New("Not found chan to communicate"),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if err := dealDelta(test.context, test.deviceID, test.payload); !reflect.DeepEqual(err, test.errWant) {
+				t.Errorf("DTManager.TestDealDelta() case failed: got = %+v, Want = %+v", err, test.errWant)
+			}
+		})
+	}
+}
+
+// TestDealSyncResult is function to test dealSyncResult
+func TestDealSyncResult(t *testing.T) {
+	twinKey := make(map[string]*dttype.MsgTwin)
+	tests := []struct {
+		name        string
+		context     *dtcontext.DTContext
+		deviceID    string
+		baseMessage dttype.BaseMessage
+		twin        map[string]*dttype.MsgTwin
+		errWant     error
+	}{
+		{
+			name:        "TestDealSyncResult(): Case 1",
+			context:     &dtcontext.DTContext{},
+			deviceID:    "DeviceA",
+			baseMessage: dttype.BaseMessage{},
+			twin:        twinKey,
+			errWant:     errors.New("Not found chan to communicate"),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if err := dealSyncResult(test.context, test.deviceID, test.baseMessage, test.twin); !reflect.DeepEqual(err, test.errWant) {
+				t.Errorf("DTManager.TestDealSyncResult() case failed: got = %v, Want = %v", err, test.errWant)
+			}
+		})
+	}
+}
+
+// TestDealDocument is function to test dealDocument
+func TestDealDocument(t *testing.T) {
+	twinDocKey := make(map[string]*dttype.TwinDoc)
+	tests := []struct {
+		name         string
+		context      *dtcontext.DTContext
+		deviceID     string
+		baseMessage  dttype.BaseMessage
+		twinDocument map[string]*dttype.TwinDoc
+		errWant      error
+	}{
+		{
+			name:         "TestDealDocument(): Case 1",
+			context:      &dtcontext.DTContext{},
+			deviceID:     "DeviceA",
+			baseMessage:  dttype.BaseMessage{},
+			twinDocument: twinDocKey,
+			errWant:      errors.New("Not found chan to communicate"),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if err := dealDocument(test.context, test.deviceID, test.baseMessage, test.twinDocument); !reflect.DeepEqual(err, test.errWant) {
+				t.Errorf("DTManager.TestDealDocument() case failed: got = %v, Want = %v", err, test.errWant)
+			}
+		})
+	}
+}
+
+// TestDealGetTwin is function to test DealGetTwin
+func TestDealGetTwin(t *testing.T) {
```

Comment:
we already have a test for the dealTwinGet function which calls DealGetTwin method inside. Why do we have to again test this function ? It's better we consolidate all such tests. also if DealGetTwin . also this method DealGetTwin could be made private.